src/components/RoiCalculatorModal/index.tsx

```diff
@@ -0,0 +1,493 @@
+  width: 100%;
+
+  & > button {
+    width: 100%;
+  }
+
+  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
+`
+
+const RoiInputContainer = styled(Flex)`
+  position: relative;
+  & > input {
+    padding-left: 28px;
+  }
+  &:before {
+    position: absolute;
+    content: '$';
+    color: ${({ theme }) => theme.colors.textSubtle};
+    left: 16px;
+    top: 8px;
+  }
+`
+
+const InvestmentInputContainer = styled.div`
+  position: relative;
+
+  & input {
+    padding-right: 34px;
+  }
+
+  &:before {
+    position: absolute;
+    content: 'USD';
+    color: ${({ theme }) => theme.colors.textSubtle};
+    right: 16px;
+    top: 21px;
+  }
+`
+
+const BulletList = styled.ul`
+  list-style-type: none;
+  margin: 0;
+  padding: 0;
+
+  li {
+    margin: 0;
+    padding: 0;
+  }
+
+  li::before {
+    content: '•';
+    margin-right: 4px;
+  }
+
+  li::marker {
+    font-size: 12px;
+    color: ${({ theme }) => theme.colors.textSubtle};
+  }
+`
+
+const compoundingIndexToFrequency = {
+  0: 2,
+  1: 1,
+  2: 0.142857142, // once every 7 days
+  3: 0.033333333, // once every 30 days
+}
+
+const getRoiData = ({ investment, interest, tokenPrice, isEditingRoi, expectedRoi, showBasedOneExpectedRoi }) => {
+  if (isEditingRoi) {
+    const expectedTokenRoi = parseFloat(expectedRoi) / tokenPrice
+    return {
+      usd: 0,
+      percentage: 0,
+      token: Number.isNaN(expectedTokenRoi) ? 0 : expectedTokenRoi,
+    }
+  }
+  if (showBasedOneExpectedRoi) {
+    const tokenRoi = expectedRoi / tokenPrice
+    const percentage = getRoi({
+      amountEarned: expectedRoi,
+      amountInvested: investment,
+    })
+    return {
+      usd: expectedRoi ?? '0.00',
+      percentage: percentage ? percentage.toFixed(2) : '0.00',
+      token: tokenRoi ? tokenRoi.toFixed(3) : '0.000',
+    }
+  }
+  if (Number.isNaN(interest)) {
+    return {
+      usd: 0,
+      percentage: 0,
+      token: 0,
+    }
+  }
+  return {
+    usd: (interest * tokenPrice).toFixed(2),
+    percentage: getRoi({
+      amountEarned: interest * tokenPrice,
+      amountInvested: investment,
+    }).toFixed(2),
+    token: interest,
+  }
+}
+
+const RoiCalculatorModal: React.FC<RoiCalculatorModalProps> = ({
+  onDismiss,
+  earningTokenPrice,
+  apr,
+  displayApr,
+  linkLabel,
+  linkHref,
+  stakingTokenBalance,
+  stakingTokenSymbol,
+  stakingTokenPrice,
+  multiplier,
+  earningTokenSymbol = 'CAKE',
+  roundingDecimals = 2,
+  autoCompoundFrequency = 0,
+  performanceFee = 0,
+  isFarm = false,
+}) => {
+  const { t } = useTranslation()
+  const { account } = useActiveWeb3React()
+  const [stakeDuration, setStakeDuration] = useState(3)
+  const [stakingTokenUSDValue, setStakingTokenUSDValue] = useState('')
+  const [compoundingFrequency, setCompoundingFrequency] = useState(1)
+  const [isEditingRoi, setIsEditingRoi] = useState(false)
+  const [showBasedOneExpectedRoi, setShowBasedOnExpectedRow] = useState(false)
```

Comment:
`setShowBasedOnExpectedRow` typo on Roi?

Reply:
fixed in ada53abe3d13902799f0f00433306f778310432d

Reply:
ah, lol. I thought you said typo on "One" -> "On"
just noticed Row -> Roi. Will change in next commit